--- a/src/components/AlbumList.js
+++ b/src/components/AlbumList.js
@@ -1,7 +1,15 @@
 import { API } from 'aws-amplify';
 import React, { useEffect, useState } from 'react';
 import { NavLink } from 'react-router-dom';
-import { Dimmer, Header, List, Loader, Segment } from 'semantic-ui-react';
+import {
+  Button,
+  Dimmer,
+  Header,
+  Icon,
+  List,
+  Loader,
+  Segment
+} from 'semantic-ui-react';
 
 import { listAlbums } from '../graphql/queries';
 import useAuth from '../useAuth';
@@ -10,22 +18,29 @@ export default function AlbumList() {
   const { owner } = useAuth();
   const [albums, setAlbums] = useState([]);
   const [isLoading, setIsLoading] = useState(true);
+  const [nextToken, setNextToken] = useState();
+  const [pageTokens, setPageTokens] = useState([]);
+  const pageToken = pageTokens[pageTokens.length - 1];
 
   useEffect(() => {
     setIsLoading(true);
 
     API.graphql({
       authMode: owner ? 'AMAZON_COGNITO_USER_POOLS' : 'AWS_IAM',
-      query: listAlbums
+      query: listAlbums,
+      variables: {
+        nextToken: pageToken
+      }
     })
       .then(payload => {
+        setNextToken(payload.data.listAlbums.nextToken);
         setAlbums(payload.data.listAlbums.items);
         setIsLoading(false);
       })
       .catch(payload => {
         throw new Error(payload.errors[0].message);
       });
-  }, [owner]);
+  }, [owner, pageToken]);
 
   return (
     <Segment.Group>
@@ -50,6 +65,28 @@ export default function AlbumList() {
           </List>
         )}
       </Segment>
+
+      <Segment clearing secondary>
+        <Button
+          disabled={!pageToken}
+          icon
+          labelPosition="left"
+          onClick={() => setPageTokens(pageTokens.slice(0, -1))}
+        >
+          Previous <Icon name="left arrow" />
+        </Button>
+
+        <Button
+          disabled={!nextToken}
+          icon
+          labelPosition="right"
+          floated="right"
+          onClick={() => setPageTokens([...pageTokens, nextToken])}
+          primary
+        >
+          Next <Icon name="right arrow" />
+        </Button>
+      </Segment>
     </Segment.Group>
   );
 }