--- a/src/components/AlbumDetails.js
+++ b/src/components/AlbumDetails.js
@@ -19,6 +19,7 @@ import {
 
 import DeleteAlbum from './DeleteAlbum';
 import PhotoList from './PhotoList';
+import PhotosUploader from './PhotosUploader';
 import { getAlbum } from '../graphql/queries';
 import { updateAlbum } from '../graphql/mutations';
 import useAuth from '../useAuth';
@@ -145,6 +146,7 @@ function AlbumDetails({ album }) {
             </Header.Subheader>
           )}
         </Header>
+        <PhotosUploader album={album} />
       </Segment>
 
       <Segment data-test="album-content" tertiary>