--- a/src/App.js
+++ b/src/App.js
@@ -4,9 +4,11 @@ import { Grid } from 'semantic-ui-react';
 import 'semantic-ui-css/semantic.min.css';
 
 import AlbumList from './components/AlbumList';
+import AlbumDetails from './components/AlbumDetails';
 import Login from './components/Login';
 import Navigation from './components/Navigation';
 import NewAlbum from './components/NewAlbum';
+import PhotoBox from './components/PhotoBox';
 
 import Amplify from 'aws-amplify';
 import awsconfig from './aws-exports';
@@ -32,6 +34,13 @@ export default function App(props) {
               </div>
             )}
           />
+
+          <Route
+            path="/albums/:albumId"
+            render={props => <AlbumDetails id={props.match.params.albumId} />}
+          />
+
+          <Route component={PhotoBox} path="/albums/:albumId/photos/:photoId" />
         </Grid.Column>
       </Grid>
     </BrowserRouter>